disprel_split.py

```diff
@@ -0,0 +1,169 @@
+
+
+parser = argparse.ArgumentParser(description='Plasma Dispersion Processor')
+parser.add_argument('--i', default=None, type=str, help='Path to E-field data')
+parser.add_argument('--periodic', default=True, type=bool, help='Period System in Y')
+parser.add_argument('--yLoc', default=4, type=int, help='Choose Y location in bounded system')
+parser.add_argument('--sN', default=4, type=int, help='No. of chuncks you want to split the processed file')
+parser.add_argument('--plot', default=True, type=bool, help='Plot the figure')
+
+args		= parser.parse_args()
+folder		= args.i
+periodic	= args.periodic
+yLoc		= args.yLoc	#8 # Choose y location
+splitNos	= args.sN #No. of chunks
+plot		= args.plot
+
+runName 	= os.path.basename(os.path.dirname(folder))
+print(runName)
+files 		= glob(pjoin(folder, 'Ex_*.dat'))
+savedir 	= pjoin(folder, '../',runName+'_processed/') #directory for processed data
+print(savedir)
+
+pat = re.compile('\d+')
+files.sort(key=lambda x: int(pat.findall(x)[-1]))
+
+#Split flie list into multiple chunks
+cSize = int(len(files)/splitNos) #Chunck size
+files_part = [files[i:i + cSize] for i in range(0, len(files), cSize)]
+# READ TEMPORAL GRID
+
+vars = parse_xoopic_input(pjoin(folder, '..', 'input.inp'))
+dt = vars['timeStep']
+
+n = np.array([int(pat.findall(a)[-1]) for a in files])
+n += 1 # XOOPIC diagnostics is actually off by one
+t = n*dt
+Nt = len(t)
+
+if  os.path.exists(savedir) and os.path.exists(savedir+'pro_files_part_0.dat'):
+	print('processed data exists. Loading data ...')
+	fp = []
+	for i in range(splitNos):
+	    ftemp = np.loadtxt(savedir+"pro_files_part_%d"%i+".dat", unpack=True)
+	    print(savedir+"pro_files_part_%d"%i+".dat")
+	    ftemp= np.array(ftemp.T)
+	    print("Shape of loaded data f1: ",ftemp.shape)
+	    fp.append(ftemp)
+	fp = np.array(fp)
+	print("Shape of final data fp: ",fp.shape)
+	# fp = np.average(fp, axis=0)
+	# fp=np.vstack((fp[0,:,:],fp[1,:,:],fp[2,:,:],fp[3,:,:]))
+	fp = fp.reshape(-1,len(fp[0,0,:]))
+	print("Shape of final data vstack fp: ",fp.shape)
+	x = np.loadtxt(savedir+"x.dat", unpack=True)
+	print("Shape of loaded data x",x.shape)
+else:
+	print('processed data does not exist. Wait, processing data ...')
+	os.mkdir(savedir)
+	fp = []# READ SPATIAL GRID
+	for i in range(splitNos):
+		data = np.loadtxt(files_part[i][0])
+		# print("DATA shape: ",len(files_part[i]))
+		x, y = data[:,2:4].T
+		Ny = len(np.where(x==x[0])[0])
+		Nx = len(x)//Ny
+		x = x.reshape(Nx, Ny)
+		y = y.reshape(Nx, Ny)
+
+		# READ FIELD
+
+		f = np.zeros((len(files_part[i]), Nx, Ny))
+
+		for j, fi in enumerate(tqdm(files_part[i])):
+			data = np.loadtxt(fi)
+			f[j] = data[:,-1].reshape(Nx, Ny)
+		print('Shape of raw f: ',f.shape)
+		# Remove y
+		#f = np.average(f, axis=2)
+		if periodic == False:
+			f = f[:,:,yLoc-4:yLoc+4]
+		f = np.average(f, axis=2)
+		fp.append(f)
+		x = np.average(x, axis=1)
+		print('Shape of f: ',f.shape)
+		np.savetxt(savedir+"pro_files_part_%d"%i+".dat",f,fmt='%.8f')
```

Comment:
A danger of `%.8f` is that if you use this function for very small field values, 8 decimals may not be enough, or that you may lose a lot of precision. Exponential notation (`e` instead of `f`) is better in this respect, but less readable. I like `g`, because it only uses exponential notation if it has to.

You may also want to be aware of `np.savez_compressed` which saves compressed binary files. It's not necessarily better, but it's a trade-off. Readability by humans and third-party software, versus a slightly smaller file size.

Reply:
In the new version, I have replaced .dat with .npz. Thanks for introducing this.